crates/rolldown_common/src/inner_bundler_options/types/treeshake.rs

```diff
@@ -68,6 +97,17 @@ impl ModuleSideEffects {
       }
       ModuleSideEffects::Boolean(false) => Some(false),
       ModuleSideEffects::Boolean(true) => None,
+      ModuleSideEffects::Function(_) => unreachable!(),
```

Comment:
I don't put them in one `resolve` function because this will force all resolve type into `async` as we all know `async` function itself has none trivial overhead, also since `sideEffects` will call `modules.len()` times.